src/Routing/Route/DashedRoute.php

```diff
@@ -49,31 +50,47 @@ public function parse($url)
         if (!$params) {
             return false;
         }
-        if (!empty($params['controller'])) {
-            $params['controller'] = Inflector::camelize(str_replace(
-                '-',
-                '_',
```

Comment:
Incidentally we could change all the inflector methods to accept a replacement char, instead of assuming it's '_' i.e. enable:

    $params['controller'] = Inflector::camelize($params['controller'], '-');
   
Which would alone prevent some bad inputs having valid output.